Microsoft/vscode-docs-authoring
# Update unit test suite

1. Update **npm run test** to run full test suite (both core extension functionality and custom linting rules.
1. Add new test command, **test-core**, to run unit tests for 'native' extension functionality.
1. Move generateTimestamp function to within extension.ts functions to unblock core extension unit tests.
1. Fix linting errors in core extension unit tests.

Testing

1. Pull latest code and checkout jamarw/run-full-test-suite branch.
1. Open terminal and navigate to docs-markdown directory in local clone.
1. Run **npm-install**.
1. Run **npm run vscode:prepublish** to compile code.
1. Run **npm run test-rules** (this command should launch four unit tests for markdownlint custom rules).
1. Run **npm run test-core** (this command should launch 23 unit tests for core extension functionality i.e. formatting).
1. Run **npm run test** (this should launch both core and custom rule unit tests).
